2 commit, создание ветки dev, добавлен базовый функционал для проверки

SergeiMakrushin opened the pull request.

Commit:
2 commit, создание ветки dev, добавлен базовый функционал для проверки

VGlyva committed:
Тестовая ветка

Akter19rus committed:
Add files via upload

Картинка для добавления в Readme проекта

Akter19rus committed:
Create README.md

Написан небольшой README

Akter19rus committed:
Merge pull request #6 from SergeiMakrushin/feature-VS

Создание README

Akter19rus committed:
Update README.md

Изменил ссылку в README для картинки preview

19Evgen78 committed:
подключен lombok

violoniste committed:
fix pom javax.persistence-api version

violoniste committed:
clear pom

Commit:
Настроено подключение к MySql, доработан класс ChatUser (убраны конструторы)

violoniste committed:
Merge branch 'master' into dev

# Conflicts:
#	README.md

Commit:
Merge branch 'feature-SM' into dev

# Conflicts:
#	pom.xml

Commit:
Merge remote-tracking branch 'origin/dev' into dev

Akter19rus committed:
Создан абстрактный класс животных
Созданы 2 подкласса с моделями собаки и кошки
Созданы таблицы в базе данных для этих сущностей

Akter19rus committed:
Merge pull request #22 from SergeiMakrushin/creatingModelsForDogAndCat

Создан абстрактный класс животных

Commit:
Сделаны контроллеры, оставил один класс животного и пользователя

19Evgen78 committed:
создан класс avatar

19Evgen78 committed:
Классы: InfoBot,ShelterAnimals,ChatUser,Avatar. Покрыты JavaDoc.

Commit:
API покрыт комментариями

Commit:
Merge branch 'feature-SM' into dev

kgskypro commented on pom.xml:
```diff
@@ -32,6 +46,66 @@
+<!--		</dependency>-->
+		<dependency>
+			<groupId>org.glassfish.jaxb</groupId>
+			<artifactId>jaxb-core</artifactId>
```
Для каких целей решили использовать jaxb?

kgskypro commented on src/main/java/com/skypro/animalShelterInfoBot/controller/AnimalController.java:
```diff
@@ -0,0 +1,83 @@
+@RestController
+@RequestMapping("/animals")
+public class AnimalController {
+    @Autowired
```
Лучше убрать @Autowired и сделать внедрение зависимости через конструктор

kgskypro commented on src/main/java/com/skypro/animalShelterInfoBot/controller/AnimalController.java:
```diff
@@ -0,0 +1,83 @@
+import java.util.List;
+
+@RestController
+@RequestMapping("/animals")
```
Необходимо добавить swagger описание контроллера

kgskypro commented on src/main/java/com/skypro/animalShelterInfoBot/controller/UserController.java:
```diff
@@ -0,0 +1,110 @@
+@RequestMapping("/user")
+@RestController
+public class UserController {
+    @Autowired
```
Лучше убрать @Autowired и сделать внедрение зависимости через конструктор

kgskypro commented on src/main/java/com/skypro/animalShelterInfoBot/controller/UserController.java:
```diff
@@ -0,0 +1,110 @@
+
+@RequestMapping("/user")
+@RestController
+public class UserController {
```
Необходимо добавить swagger описание контроллера

kgskypro commented on src/main/java/com/skypro/animalShelterInfoBot/model/Avatar.java:
```diff
@@ -0,0 +1,25 @@
+import jakarta.persistence.*;
+import lombok.*;
+
+    /**
```
Поехало форматирование кода, лишние отступы

kgskypro commented on src/main/java/com/skypro/animalShelterInfoBot/model/animals/ShelterAnimals.java:
```diff
@@ -0,0 +1,38 @@
+
+
+@Builder
+@Data
```
`@Data` – это сокращенная аннотация, сочетающая возможности `@ToString, @EqualsAndHashCode, @Getter @Setter и @RequiredArgsConstructor`

То есть нужно либо оставить `@Data` и убрать `@Setter` и `@Getter`, либо наоборот

kgskypro commented on src/main/java/com/skypro/animalShelterInfoBot/model/animals/ShelterAnimals.java:
```diff
@@ -0,0 +1,38 @@
+@Getter
+@Entity
+
+    /**
```
Java Doc съехал (лишние пробелы) и java doc должен быть размещен до аннотаций

kgskypro commented on src/main/java/com/skypro/animalShelterInfoBot/model/human/ChatUser.java:
```diff
@@ -0,0 +1,50 @@
+import lombok.*;
+
+    /**
+    * Модель таблицы, куда будут попадать все первоначально общающиеся с ботом.
```
Лишние пробелы

19Evgen78 committed:
В моделях исправлен Java Doc, убрана аннотация @Data.

19Evgen78 committed:
В классе InfoBot исправлен Java Doc.

Commit:
Заменил @Autowired на конструктор, добавил описание контроллеров

19Evgen78 committed:
Merge branch 'dev' into feature-EK

SergeiMakrushin committed:
Merge pull request #24 from SergeiMakrushin/feature-EK

Исправлены аннотации , доработан JavaDoc

Commit:
Добавлены идентификаторы доступа

Commit:
Merge remote-tracking branch 'origin/dev' into dev

19Evgen78 committed:
Создан сервис AnimalService.
Переделан контроллер AnimalController.

19Evgen78 committed:
В сервисе AnimalService добавлен метод getAnimalsPagination.
 Доделан контроллер AnimalController.

19Evgen78 committed:
В сервисе AnimalService добавлен метод getAnimalsPagination.
 Доделан контроллер AnimalController.

19Evgen78 committed:
Сделан сервис UserService.
Доделан контроллер UserController.

19Evgen78 committed:
Создан репозиторий AnimalRepository.
Создан репозиторий UserRepository.

19Evgen78 committed:
Переделаны сервисы.

VGlyva committed:
Добавил комменты на код ревью

19Evgen78 committed:
Сделаны контроллеры.
Всервисах переделан методы удаления.

Commit:
Добавлен liquibase

19Evgen78 committed:
В сервисах и контроллерах добавлены методы редактирования.

19Evgen78 committed:
AnimalController покрыт тестами.

19Evgen78 committed:
Доработанны контроллеры

19Evgen78 committed:
Merge branch 'feature-EK' into dev

# Conflicts:
#	src/main/java/com/skypro/animalShelterInfoBot/controller/AnimalController.java
#	src/main/java/com/skypro/animalShelterInfoBot/model/animals/Animal.java
#	src/main/java/com/skypro/animalShelterInfoBot/services/AnimalService.java
#	src/main/java/com/skypro/animalShelterInfoBot/services/UserService.java

19Evgen78 committed:
Добавлены исключения в сервисы.

Commit:
Merge branch 'feature-EK' into dev

# Conflicts:
#	src/main/java/com/skypro/animalShelterInfoBot/services/AnimalService.java

Commit:
изменен метод processingTextAndCallbackQuery в BotService

Commit:
Debug

Commit:
Debug

Commit:
Запустил два метода: "получить информацию о приюте" и "адрес и время работы"

Commit:
Merge branch 'feature-SM' into dev

# Conflicts:
#	src/main/java/com/skypro/animalShelterInfoBot/configuration/InfoBotConfiguration.java
#	src/main/java/com/skypro/animalShelterInfoBot/controller/AnimalController.java
#	src/main/java/com/skypro/animalShelterInfoBot/controller/UserController.java

Commit:
Запустил два метода: "получить информацию о приюте" и "адрес и время работы"

19Evgen78 committed:
Сделан сервис Avatar и контроллур Avatar.

Commit:
Аватар

19Evgen78 committed:
Сделан сервис Avatar и контроллур Avatar.

19Evgen78 committed:
Сделан сервис Avatar и контроллур Avatar.

19Evgen78 committed:
Сделан сервис Avatar и контроллур Avatar.

19Evgen78 committed:
Сделан сервис Avatar и контроллур Avatar.

Commit:
1

Commit:
2

Commit:
1

Commit:
3

Commit:
Merge branch 'feature-EK' into dev

# Conflicts:
#	src/main/java/com/skypro/animalShelterInfoBot/model/animals/Animal.java
#	src/main/java/com/skypro/animalShelterInfoBot/model/avatar/Avatar.java
#	src/main/java/com/skypro/animalShelterInfoBot/repositories/AvatarRepository.java
#	src/main/resources/application.properties

Commit:
4

19Evgen78 committed:
AvatarController покрыт комментариями

19Evgen78 committed:
Добавлены методы в BotService

19Evgen78 committed:
Сделана связь между Animal и ChatUser

Commit:
логика метода позвать волонтера

violoniste committed:
Доработка документации BotService

violoniste committed:
Рефакторринг определения команд

violoniste committed:
Merge remote-tracking branch 'origin/feature-SM' into refactoring

violoniste committed:
Перенос названия кнопок в константы

violoniste committed:
Комнады вынесены в константы

Akter19rus committed:
создание постоянной клавиатуры перенесен в BotService.settingSendMessage, для корректноого ответа на неизвестное сообщение

violoniste committed:
Единый switch для команд и кнопок

violoniste committed:
Merge remote-tracking branch 'origin/feature-VS' into refactoring

violoniste committed:
Убран лист списка кнопок. В создании меню используются константы.

violoniste committed:
Константы команд в TelegramBot из сервиса

violoniste committed:
Рефакторинг создания меню

Akter19rus committed:
изменена длина названия кнопок в константах, для корректного отображения в телеграме

Akter19rus committed:
Merge branch 'dev' into feature-VS

# Conflicts:
#	src/main/java/com/skypro/animalShelterInfoBot/service/bot/BotService.java

Akter19rus committed:
Изменен метод instructionAdoptionMenu
сделано изменение расположения кнопок

Akter19rus committed:
Merge pull request #44 from SergeiMakrushin/feature-VS

Feature vs

Akter19rus committed:
Сделан метод adviceDogHandlers
(Советы кинологов) текст сообщения находится в ShelterInformationDirectory

Akter19rus committed:
Сделан метод homeForChild
(дом для маленьких питомцев) текст сообщения находится в ShelterInformationDirectory

19Evgen78 committed:
UserController покрыт тестами

19Evgen78 committed:
UserController покрыт тестами

VGlyva committed:
Merge branch 'dev' into futureVG

Akter19rus committed:
Merge pull request #47 from SergeiMakrushin/feature-VS

create method homeForChild and adviceDogHandlers

VGlyva committed:
Merge branch 'dev' into futureVG

VGlyva committed:
Сделал интерфейсы. Реализовал метод контактов для кинологов

VGlyva committed:
Редактировал папки

19Evgen78 committed:
Merge branch 'feature-EK' into dev

19Evgen78 committed:
Avatar тест

VGlyva committed:
Merge remote-tracking branch 'origin/dev' into dev

Commit:
UserControllerWebMvcTest убрал конфликты

Commit:
UserControllerWebMvcTest убрал конфликты

19Evgen78 committed:
Merge branch 'dev' into feature-EK

19Evgen78 committed:
AvatarController покрыт тестами

VGlyva committed:
Создал методы "ссылки". Сделал единую папку model.

VGlyva committed:
Merge remote-tracking branch 'origin/dev' into dev

# Conflicts:
#	src/test/java/com/skypro/animalShelterInfoBot/controller/UserControllerWebMvcTest.java

VGlyva committed:
Исправил ошибки после merge

19Evgen78 committed:
Merge branch 'dev' into feature-EK

violoniste committed:
Слушатель для бот сервиса для отправки сообщений

19Evgen78 committed:
Merge branch 'dev' into feature-EK

19Evgen78 committed:
Исправлены тесты AvatarControllerWebMvcTest

Akter19rus committed:
Дописан метод getContactVolunteer
(вызов волонтера)

Akter19rus committed:
Сделан метод meetingAnimals
(правила знакомства с животным)
Сделан метод shelterTB
(ТБ на территории приюта)
Сделан метод registerPass
(Получить контакты охраны для регистрации пропуска)

Akter19rus committed:
Merge pull request #53 from SergeiMakrushin/feature-VS

creat methods meetingAnimals & shelterTB & registerPass

19Evgen78 committed:
Исправлен testDeleteAvatar

19Evgen78 committed:
Merge branch 'dev' into feature-EK

VGlyva committed:
Реализован метод "Досстать всех животных". Метод "Сохранить контакты" пока не работает.

Commit:
Часть юнит-тестов BotService

19Evgen78 committed:
Merge branch 'dev' into feature-EK

19Evgen78 committed:
AnimalService покрыт тестами.

19Evgen78 committed:
UserServiceImpl покрыт тестами.

19Evgen78 committed:
AvatarServiceImpl покрыт тестами.

Akter19rus committed:
Сделан метод leaveContact
(Оставить контакты)

я победил его :D

Akter19rus committed:
Merge pull request #62 from SergeiMakrushin/feature-VS

Сделан метод leaveContact (Оставить контакты)

Akter19rus committed:
Доработан метод leaveContact
(Оставить контакты)
Добавлена отправка уведомления волонтеру

я победил его :D

Akter19rus committed:
Merge pull request #63 from SergeiMakrushin/feature-VS

Доработан метод leaveContact

19Evgen78 committed:
Merge branch 'dev' into feature-EK

19Evgen78 committed:
В AvatarControllerWebMvcTest исправлен тест.

Akter19rus committed:
сделан метод registerUserAndWelcome
(Проверка наличия пользователя в БД, если нет - добавляем)
Если добавляем, приветствуем как новенького)

Akter19rus committed:
Merge pull request #64 from SergeiMakrushin/feature-VS

сделан метод registerUserAndWelcome

Commit:
AvatarServiceImplTest, AnimalServiceImplTest

Commit:
Merge remote-tracking branch 'origin/feature-EK' into feature-EK

Commit:
testDownloadPreview

Commit:
Merge branch 'feature-SM' into dev

Commit:
public SendMessage checkingTextForContacts in BotServiceImpl

Commit:
public SendMessage checkingTextForContacts in BotServiceImpl

19Evgen78 committed:
В BotServiceImplTest добавлен один тест.

19Evgen78 committed:
В BotServiceImplTest добавлены три три теста.

19Evgen78 committed:
В BotServiceImplTest добавлен один тест.

19Evgen78 committed:
В BotServiceImplTest добавлены три теста.

19Evgen78 committed:
TelegramBot покрыт тестами.

19Evgen78 committed:
InfoBotConfiguration покрыт тестами.

Akter19rus committed:
изменен метод getAllDogAndCat
(показать всех животных) - теперь показывает тех животных, которые находятся в отделе в который зашли.
создан метод checkingTextForColor
(обрабатывает текст сообщения ) - если находит соответствие с цветом, и находит животного с этим цветом, отправляет пользователю, если не находит отправляет NOTFOUNDTEXT

Akter19rus committed:
Merge pull request #65 from SergeiMakrushin/feature-VS

изменен метод getAllDogAndCat

19Evgen78 committed:
AnimalShelterInfoBotApplication покрыт тестами.

19Evgen78 committed:
InfoBotInitializer покрыт тестами.

19Evgen78 committed:
Все модели покрыты тестами.

Akter19rus committed:
изменен вывод собак и кошек, теперь показывает без квадратных скобок

Akter19rus committed:
Merge pull request #66 from SergeiMakrushin/feature-VS

изменен вывод собак и кошек, теперь показывает без квадратных скобок

Commit:
Поле TapeDogAndCat

Akter19rus committed:
создан метод поиск животных по диапазону возраста

Akter19rus committed:
Merge pull request #67 from SergeiMakrushin/feature-VS

создан метод поиск животных по диапазону возраста

VGlyva committed:
Реализован метод поиск по породе (требуется рефакторинг)

Akter19rus committed:
создан метод поиск животных по породе
так же навешаны флаги на все обработки сообщений
теперь методы обработки будут работать только тогда, когда нажали на соответствующую кнопку.
Снятие флагов происходит когда пользователь нажмет на любую кнопку

Akter19rus committed:
Merge pull request #71 from SergeiMakrushin/futureVG

Future vg

Commit:
Merge branch 'dev' into feature-SM

# Conflicts:
#	src/main/java/com/skypro/animalShelterInfoBot/bot/BotServiceImpl.java

Akter19rus committed:
изменены методы
checkingTextForColor - (найти по цвету),
checkingTextForBreed - (найти по породе),
теперь методы поиска просто обрабатывают текст входящего сообщения и если текст совпал со значением поля животного, то отвечает информацией по этим животным.
так же убраны паттерны по которым искалось соответствие, так как теперь используются булевые флаги

Akter19rus committed:
Merge pull request #72 from SergeiMakrushin/feature-VS

изменены методы -

Commit:
Метод отправки отчета

Commit:
Метод отправки отчета доработка, добавлены перегруженные методы

SergeiMakrushin committed:
Метод отправки отчета, частично заработал, не сохраняет в базу

Akter19rus committed:
создан метод бронирования животного и сохранения(присвоение) животному пользователя checkingTextForIdAnimalsAndMerge

Akter19rus committed:
Merge pull request #73 from SergeiMakrushin/feature-VS

создан метод бронирования животного и сохранения(присвоение) животном…

Commit:
Метод отправки отчета, сохраняет в базу, необходим рефакторинг

Commit:
Метод отправки отчета, сделан рефакторинг

Commit:
Merge branch 'feature-SM' into dev

# Conflicts:
#	src/main/java/com/skypro/animalShelterInfoBot/bot/BotServiceImpl.java

Commit:
Метод отправки отчета, сделан рефакторинг

19Evgen78 committed:
Исправленны тесты в TelegramBotTest.

19Evgen78 committed:
Исправленны тесты в TelegramBotTest.

VGlyva committed:
Создан тест testInputMsg

VGlyva committed:
Merge remote-tracking branch 'origin/dev' into dev

# Conflicts:
#	src/test/java/com/skypro/animalShelterInfoBot/bot/BotServiceImplTest.java

VGlyva committed:
Создан тест testInputMsg

Commit:
Создание таблиц в liquibase
